### PR TITLE
Pci bus ID now printed for GPU cards

### PR DESCRIPTION
**Pull request by notoraptor**
This update allows Theano to print PCI bus ID for the GPU card (related to issue #5176 ).

This will works with CUDA devices and with the last version of libgpuarray ( see this pull request: https://github.com/Theano/libgpuarray/pull/283 ).

OpenCL devices are currently not supported (an "unsupported" message should be printed).

Tested on both Python 2 and Python 3. On my computer:

NVIDIA infos:

```nvidia-smi```:

```
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 370.28                 Driver Version: 370.28                    |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce 840M        Off  | 0000:0A:00.0     Off |                  N/A |
| N/A   51C    P0    N/A /  N/A |    267MiB /  2002MiB |      5%      Default |
+-------------------------------+----------------------+----------------------+
```

Short test with CUDA:

```THEANO_FLAGS=device=cuda python -c "import theano"```
```
Mapped name None to device cuda: GeForce 840M; PCI Bus ID: 0000:0A:00.0
```

Short test with OpenCL:

```THEANO_FLAGS=device=opencl0:0 python -c "import theano"```
```
Mapped name None to device opencl0:0: GeForce 840M; PCI Bus ID: (unsupported for device opencl0:0)
```

### CODE REVIEW

File: theano/gpuarray/__init__.py [hunk 1]
```diff
@@ -85,8 +85,16 @@ def init_dev(dev, name=None):
     # This will map the context name to the real context object.
     reg_context(name, context)
     if config.print_active_device:
-        print("Mapped name %s to device %s: %s" %
-              (name, dev, context.devname),
+        try:
+            pcibusid = context.pcibusid
+        except pygpu.gpuarray.UnsupportedException:
+            pcibusid = '(unsupported for device %s)' % dev
+        except:
```
Comment (nouiz) [inline theano/gpuarray/__init__.py]: Never do that. Always catch an exception. It can be "Exception"

File: theano/gpuarray/__init__.py [hunk 2]
```diff
@@ -85,8 +85,16 @@ def init_dev(dev, name=None):
...
+            warnings.warn('Unable to get PCI Bus ID. Please consider updating libgpuarray and pygpu.')
+            pcibusid = 'unknown'
+
+        print("Mapped name %s to device %s: %s; PCI Bus ID: %s" %
+              (name, dev, context.devname, pcibusid),
```
Comment (abergeron) [inline theano/gpuarray/__init__.py]: I would prefer it if the PCI bus id was on another line

Comment (notoraptor) [inline theano/gpuarray/__init__.py]: Updated:

```
(python2) notoraptor:theano$ THEANO_FLAGS=device=cuda python -c "import theano"
Mapped name None to device cuda: GeForce 840M
PCI Bus ID: 0000:0A:00.0
(python2) notoraptor:theano$ THEANO_FLAGS=device=opencl0:0 python -c "import theano"
Mapped name None to device opencl0:0: GeForce 840M
PCI Bus ID: (unsupported for device opencl0:0)
```

File: theano/gpuarray/__init__.py (hunk 1 above)
Comment (notoraptor) [inline theano/gpuarray/__init__.py]: Done !

**Pull request merged**